src/tools/auth0/handlers/prompts.ts

```diff
@@ -186,9 +186,17 @@ export default class PromptsHandler extends DefaultHandler {
   }
 
   async getCustomTextSettings(): Promise<AllPromptsByLanguage> {
-    const supportedLanguages = await this.client.tenant
-      .getSettings()
-      .then(({ enabled_locales }) => enabled_locales);
+    const supportedLanguages = await (async () => {
+      const enabledLocales = await this.client.tenant
+        .getSettings()
+        .then(({ enabled_locales }) => enabled_locales);
+
+      if (!Array.isArray(enabledLocales)) {
+        return []; // For reasons unknown, enabled_locales is sometimes not an array. See: ESD-20252
+      }
```

Comment:
This is the core change of the PR. Not elegant by any means and a bit of a nasty precedent to set, but it will solve the customer's crashes.